Update readme with app features (#694)
anitab-org/mentorship-android: README.md

 * **develop** This contains the latest code. All the contributing PRs must be sent to this branch. When we want to release the next version of the app, this branch is merged into the `master` branch.

 * **apk** This branch contains the apks for the code in the develop branch. The apks are automatically updated when a commit is pushed to `develop` branch.
 

> ## Features

Reviewer: Maybe this session would be better before setting up the project, also without a header.
Since this is part of the introduction to the app

  
The Application has the following features:

 * Users can **find** and **search** other **members** and check their profile.

>  * Users can **send request** to other members as a mentor or mentee according to their interests.

Reviewer: ```suggestion
 * A member can **send request** to other members as a mentor or mentee according to their interests.
```

 * One can check the status of their **pending, past and rejected requests**.
 * The user can check their **current relationships**.

>  * Mentors can add **tasks** in a mentorship relation.

Reviewer: ```suggestion
 * Mentors and mentees can create, update, visualize and delete **tasks** in their current mentorship relation.
```

>  * Users can check all the tasks in a mentorship relation.

Reviewer: delete this one and add this as part of the first above one. I will leave an example

>  * Functionality to **update profile**.

Reviewer: ```suggestion
 * Users can **update their profile**.
```

>  * Users also have the privilege to see their **recent achievements**.

Reviewer: ```suggestion
 * Users can also see their **recent achieved tasks**.
```

Changed to achieved tasks, for people to understand that achievements = tasks completed

>  * Functionality to **give feedback**.

Reviewer: not yet, since there is not functionality integrated (so I would consider it as not implemented)

>  * Other functionalities like **change password, delete account and about us**.

Reviewer: Maybe remove this line, because the delete feature is not implemented yet I think 🤔 the option exists but does not trigger a request. Also, we have to revisit how this works on the backend.
